catalog/src/main/java/org/killbill/billing/catalog/caching/DefaultOverriddenPlanCache.java

```diff
@@ -89,14 +85,23 @@ public DefaultPlan getOverriddenPlan(final String planName, final StandaloneCata
         args[1] = catalog;
 
         final CacheLoaderArgument argument = new CacheLoaderArgument(irrelevant, args, context);
-        final DefaultPlan defaultPlan = (DefaultPlan) cacheController.get(planName, argument);
+        final String planNameVersion = getPlanNameVersion(planName, catalog);
+        final DefaultPlan defaultPlan = (DefaultPlan) cacheController.get(planNameVersion, argument);
         defaultPlan.initialize(catalog);
         return defaultPlan;
     }
 
     @Override
     public void addDryRunPlan(final String planName, final Plan plan) {
-        cacheController.putIfAbsent(planName, plan);
+        final String planNameVersion = getPlanNameVersion(planName, plan.getCatalog());
```

Comment:
Two questions:
1. Why do we cache these dry-run plans in the main long-term cache?
2. What's the cache memory footprint impact? Could we end up creating a lot more entries now?

For large catalogs, these can be 25MB+, so we should ensure we only cache what's really needed.

Reply:
> Why do we cache these dry-run plans in the main long-term cache?

Added another layer of cache would complexify the code, I suppose that is (was) the reason.

> What's the cache memory footprint impact? Could we end up creating a lot more entries now?

Exactly the same number, because each dry-run entry would be its own entry anyways, allowing 2 do 2 dry-runs in a row with different attributes. Only the key changed. 

Also, there is should be a limit in the size of the cache and some possibility to configure cache eviction.